tests/runtests.pl

```diff
@@ -2744,15 +2744,21 @@ sub cleardir {
     my $file;
 
     # Get all files
-    opendir(DIR, $dir) ||
+    opendir(my $dh, $dir) ||
         return 0; # can't open dir
-    while($file = readdir(DIR)) {
-        if(($file !~ /^\./)) {
-            unlink("$dir/$file");
+    while($file = readdir($dh)) {
+        if(($file !~ /^(\.|\.\.)\z/)) {
```

Comment:
curious, whats motivated this change in runtests.pl ?

Reply:
because we do tests that create directories in the log dir (including in this PR), but before this particular fix we didn't delete such directories!

Reply:
The dot removal thing was landed separately in #5838